Failing case (before fix): https://community-tc.services.mozilla.com/tasks/EL3pj6KRSLONLj8EdPGw8g/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FEL3pj6KRSLONLj8EdPGw8g%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

```
E     {u'message': None,
E     u'name': u"assert_true(true, '3')",
E     u'properties': {},
E     u'status_string': u'PASS'},
E     {u'message': None,
E     u'name': u"assert_true(true, '3') 1",
E     u'properties': {},
E     u'status_string': u'PASS'},
E     {u'message': None,
E     -                         u'name': u"assert_true(true, '3') 1",
E     ?                                                           ^
E     +                         u'name': u"assert_true(true, '3') 2",
E     ?                                                           ^
E     u'properties': {},
E     u'status_string': u'PASS'},
```

--- a/resources/testharness.js
+++ b/resources/testharness.js
@@ -541,14 +541,15 @@ policies and contribution forms [3].
                 trimmed = trimmed.replace(/^([^;]*)(;\s*)+$/, "$1");
 
                 if (trimmed) {
-                    // add a suffix if we already have this string
+                    let name = trimmed;
                     if (seen_func_name[trimmed]) {
-                        let current_count = seen_func_name[trimmed]++;
-                        trimmed = trimmed + " " + current_count;
+                        // This subtest name already exists, so add a suffix.
+                        name += " " + seen_func_name[trimmed];
                     } else {
-                        seen_func_name[trimmed] = 1;
+                        seen_func_name[trimmed] = 0;
                     }
-                    return trimmed;
+                    seen_func_name[trimmed] += 1;
+                    return name;
                 }
             }
         }

